### DISCUSSION TITLE
How to invalidate manifest.json cache?

### DISCUSSION THREAD

**Discussion by welschmoor** [serwist/serwist#87] [Q&A]
Hi,
Everything I have entered inside `manifest.json` seems to remain forever on installed PWAs. So if I change the background color from white to some other color, redeploy, the change won't happen even after 30 hours (tested on Android and Mac).  `runtimeCaching` inside sw.ts field is missing, I don't use it (but have also tried first filtering, then passing empty array without success). Thanks!
```
{
  ....
  "theme_color": "#FFFFFF", // <- change this, but PWA color remains #FFF
  "background_color": "#FFFFFF", // <- change this, redeploy, but PWA color remains #FFF
  "start_url": "/",
   ...
}
```

I would need to force people to reinstall the app somehow for change to take effect and that's obviously bad.
My PWA knowledge is very limited. Thanks in advance!

**Comment by DuCanhGH** [comment]
@welschmoor by default, manifest.json is precached. This precached response is then reused until the next service worker install, during which we would recache URLs whose integrity/revision has changed. Do check if this value changes when you modify manifest.json. Otherwise, either there's a bug or the service worker takes too long to install. However, If your service worker takes 30 hours to reinstall, there's probably something wrong with it...
